### DISCUSSION TITLE
Using server side blazor without sticky sessions

### DISCUSSION THREAD

**Discussion by GTmAster** [dotnet/aspnetcore#25690] [General] [answered]
Currently we investigating server side Blazor and how it could be helpful for us. But we got a problem. We have ELB as a load balancer and we can't configure sticky session on it (company IT policy). So blazor connection can't even be initialized correctly, because 

1. Request POST _blazor/negotiate?negotiateVersion=1
2. Request GET _blazor?id=<connection_id>&_=<...>

go to different servers. Of course it doesn't work, because circuit is stored in memory.

Is there any technique to store connection and circuit altogether in some shared storage, redis for example?

**Comment by SteveSandersonMS** [comment, marked as answer]
No, I'm afraid this is not supported.

The underlying reason is that circuits aren't serializable, since in general they can contain arbitrary data from your own application as well as things like delegates. So, circuit data can only exist in memory.

**Reply by GTmAster** [reply]
Thanks for the answer. We will try to utilize WebAssembly blazor.
